nextcord/ui/modal.py

```diff
@@ -0,0 +1,368 @@
+        
+        self.children: List[Item] = []
+        for key, component in components.items():
+            setattr(self, key, component)
+            self.children.append(component)
+        
+        self.__weights = _ViewWeights(self.children)
+        loop = asyncio.get_running_loop()
+        self.id: str = os.urandom(16).hex()
+        self.__cancel_callback: Optional[Callable[[Modal], None]] = None
+        self.__timeout_expiry: Optional[float] = None
+        self.__timeout_task: Optional[asyncio.Task[None]] = None
+        self.__stopped: asyncio.Future[bool] = loop.create_future()
+    
+    async def __timeout_task_impl(self) -> None:
+        while True:
+            # Guard just in case someone changes the value of the timeout at runtime
+            if self.timeout is None:
+                return
+
+            if self.__timeout_expiry is None:
+                return self._dispatch_timeout()
+
+            # Check if we've elapsed our currently set timeout
+            now = time.monotonic()
+            if now >= self.__timeout_expiry:
+                return self._dispatch_timeout()
+
+            # Wait N seconds to see if timeout data has been refreshed
+            await asyncio.sleep(self.__timeout_expiry - now)
+
+    def to_components(self) -> List[Dict[str, Any]]:
+        def key(item: Item) -> int:
+            return item._rendered_row or 0
+
+        children = sorted(self.children, key=key)
+        components: List[Dict[str, Any]] = []
+        for _, group in groupby(children, key=key):
+            children = [item.to_component_dict() for item in group]
+            if not children:
+                continue
+
+            components.append(
+                {
+                    'type': 1,
+                    'components': children,
+                }
+            )
+
+        return components
+
+    def to_dict(self) -> Dict[str, Any]:
+        payload = {
+            'title': self.title,
+            'custom_id': self.custom_id,
+            'components': self.to_components(),
+        }
+        return payload
+    
+    @property
+    def _expires_at(self) -> Optional[float]:
+        if self.timeout:
+            return time.monotonic() + self.timeout
+        return None
+    
+    def add_item(self, item: Item) -> None:
+        """Adds an item to the modal.
+
+        Parameters
+        -----------
+        item: :class:`Item`
+            The item to add to the modal.
+
+        Raises
+        --------
+        TypeError
+            An :class:`Item` was not passed.
+        ValueError
+            Maximum number of children has been exceeded (25)
+            or the row the item is trying to be added to is full.
+        """
+
+        if len(self.children) > 25:
+            raise ValueError('maximum number of children exceeded')
+
+        if not isinstance(item, Item):
+            raise TypeError(f'expected Item not {item.__class__!r}')
+
+        self.__weights.add_item(item)
```

Comment:
a modal can only have up to 5 items and there is no need to calculate open space or all the other things ViewWeight.add_item does.

Reply:
I think Discord should throw an http exception if it's exceeded anyway, so it can be left unchecked possibly